### DISCUSSION TITLE
Extended MMX (Cyrix EMMI) instructions

### DISCUSSION THREAD

**Discussion by Torinde** [FEX-Emu/FEX#3576] [Ideas]
What do you think of emulating [Extended MMX (Cyrix EMMI)](https://en.wikipedia.org/wiki/List_of_discontinued_x86_instructions#Cyrix_EMMI_instructions) instructions?

- There is a [CPU flag in Linux](https://unix.stackexchange.com/questions/43539/what-do-the-flags-in-proc-cpuinfo-mean) for those: `cxmmx` Cyrix MMX extensions
- [Cyrix Extension to the Multimedia Instruction Set, Application Note 108](http://hackipedia.org/browse.cgi/Hardware/CPU/x86/Cyrix%206x86/Cyrix%20Extension%20to%20the%20Multimedia%20Instruction%20Set%2c%20Application%20Note%20108%2c%20Applies%20to%20the%20Cyrix%20MII%e2%84%a2%20CPU%20%281998-09-09%29.pdf) (some further docs and software are linked [here](https://github.com/PCBox/PCBox/issues/39#issuecomment-1147292376))
- Opcodes overlap with SSE and other subsequent instructions, so it has to be selectable/option?
- https://github.com/86Box/86Box/pull/5284

**Comment by Sonicadvance1** [comment]
Implementing the instructions likely wouldn't be difficult but since the instruction encodings overlap it's a little annoying.
It would only be worth implementing if there was some interesting application that uses them and /doesn't/ have an accelerated MMX/SSE/3DNow! path.

**Reply by Torinde** [reply]
Clear, thanks!
